.github/workflows/type-check.yml

```
@@ -0,0 +1,36 @@
+        bun-version: 1.0.11
+    - run: bun install
+
+    - name: Check out the code
+      id: type-check
+      run: |
+        bun run type-check
+        bun run lint
+    
+    - name: Report type check errors
+      if: failure() && steps.type-check.outcome == 'failure'
+      uses: actions/github-script@v6
```

actionlint (1.7.4)

the runner of "actions/github-script@v6" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)